src/mongo/MongoScrapbook.ts

```diff
@@ -281,7 +281,9 @@ class FindMongoCommandsVisitor extends MongoVisitor<MongoCommand[]> {
 		} else if (child instanceof ErrorNode) {
 			return {};
 		} else {
-			throw new Error("Unrecognized node type encountered");
+			let err = new Error(`Unrecognized node type encountered. We could not parse ${child.text}`);
```

Comment:
I don't think this is the right way to go about this. If you're call `addErrorToCommand` each time individually then you're asking for another bug like this to happen. Someone else could easily forget to do this in the future.

Plus the current code means we don't get any telemetry.

Reply:
This was meant only for a quick temporary patch.

Reply:
@StephenWeatherford I assumed that "patch" was the planned release we have for Cosmos DB next Wednesday (as a part of the app service "Connections" feature). So I had already gone ahead with the other approach: https://github.com/Microsoft/vscode-cosmosdb/pull/960

Reply:
I'd rather take enough time to make sure this is fixed properly after a quick patch.

Reply:
The screenshot shows errors thrown from functionCallContextToObject. Almost all the errors in the telemetry arrive from there. I've covered that approach below.

Reply:
I still want an all-encompassing try/catch before I approve this. Something like this: https://github.com/Microsoft/vscode-cosmosdb/commit/869dd1dae9d51a9a381312adef2f3e7cc5e1b3d2?w=1

Reply:
That would be good.

Reply:
Pushed. :)